src/app/modules/form/admin-form/admin-form.issue.controller.ts

```diff
@@ -0,0 +1,160 @@
+import { AuthedSessionData } from 'express-session'
+import { StatusCodes } from 'http-status-codes'
+import JSONStream from 'JSONStream'
+
+import {
+  ErrorDto,
+  FormIssueMetaDto,
+  FormIssueMetaQueryDto,
+} from '../../../../../shared/types'
+import { createLoggerWithLabel } from '../../../config/logger'
+import { createReqMeta } from '../../../utils/request'
+import * as AuthService from '../../auth/auth.service'
+import { ControllerHandler } from '../../core/core.types'
+import * as IssueService from '../../issue/issue.service'
+import * as UserService from '../../user/user.service'
+
+import { PermissionLevel } from './admin-form.types'
+import { mapRouteError } from './admin-form.utils'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Handler for GET /{formId}/issue.
+ * @security session
+ *
+ * @returns 200 with issue response
+ * @returns 403 when user does not have permissions to access form
+ * @returns 404 when form cannot be found
+ * @returns 410 when form is archived
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const handleGetFormIssue: ControllerHandler<
+  { formId: string },
+  FormIssueMetaDto | ErrorDto,
+  unknown,
+  FormIssueMetaQueryDto
+> = (req, res) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as AuthedSessionData).user._id
+
+  return UserService.getPopulatedUserById(sessionUserId)
+    .andThen((user) =>
+      AuthService.getFormAfterPermissionChecks({
+        user,
+        formId,
+        level: PermissionLevel.Read,
+      }),
+    )
+    .andThen(() => IssueService.getFormIssues(formId))
+    .map((response) => res.json(response))
+    .mapErr((error) => {
+      logger.error({
+        message: 'Error retrieving form issues',
+        meta: {
+          action: 'getFormIssue',
+          ...createReqMeta(req),
+          userId: sessionUserId,
+          formId,
+        },
+        error,
+      })
+      const { errorMessage, statusCode } = mapRouteError(error)
+      return res.status(statusCode).json({ message: errorMessage })
+    })
+}
+
+/**
+ * Handler for GET /{formId}/issue/download.
+ * @security session
+ *
+ * @returns 200 with issue stream
+ * @returns 403 when user does not have permissions to access form
+ * @returns 404 when form cannot be found
+ * @returns 410 when form is archived
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database or stream error occurs
+ */
+export const handleStreamFormIssue: ControllerHandler<
```

Comment:
/nit since we are returning multiple issues, does it make sense for this to be plural? (same for `handleGetFormIssue`)

Reply:
it does, but im following FormFeedback style. Should i follow the existing or add new convention? 🤔

Reply:
Technically, feedback has no plural 😆 (something like there's no foods, only food).

Reply:
done https://github.com/opengovsg/FormSG/pull/6516/commits/dad6758c7ee8e83eb6e270976ea80879b5a3c8c5